### PR TITLE
completed 8-update-articles

### PR DESCRIPTION
**Pull request by zaw-tun**
completed route PATCH /api/articles/:article_id

### CODE REVIEW

File: __tests__/api.test.js [hunk 1]
```diff
@@ -219,3 +219,39 @@ describe("/POST /api/articles/:article_id/comments", () => {
       });
   });
 });
+
+describe("/PATCH /api/articles/:article_id", () => {
+  const updateVotes = {
+    inc_votes: 10,
+  };
+  it("status:200, responds with updated article", () => {
```
Comment (nc-aimee-muter) [inline __tests__/api.test.js:227]: Nice test. You have not checked that votes can be decreased so you need another test for this endpoint

File: __tests__/api.test.js [hunk 2]
```diff
@@ -219,3 +219,39 @@ describe("/POST /api/articles/:article_id/comments", () => {
       });
   });
 });
+
+describe("/PATCH /api/articles/:article_id", () => {
+  const updateVotes = {
```
Comment (nc-aimee-muter) [inline __tests__/api.test.js:224]: Never leave test variables outside the test blocks. This object is vulnerable to mutation by all tests as it is in the scope of all tests. Redeclare this inside each test block.

File: __tests__/api.test.js [hunk 3]
```diff
@@ -219,3 +219,39 @@ describe("/POST /api/articles/:article_id/comments", () => {
...
+    return request(app)
+      .patch("/api/articles/5")
+      .send(updateVotes)
+      .expect(200)
+      .then((response) => {
+        expect(response.body.article.article_id).toBe(5);
+        expect(response.body.article.votes).toBe(10);
+      });
+  });
+  it("status: 404, responds with an appropriate status and error when given a valid but non-exisitant article_id", () => {
```
Comment (nc-aimee-muter) [inline __tests__/api.test.js:237]: Nice

File: __tests__/api.test.js [hunk 4]
```diff
@@ -219,3 +219,39 @@ describe("/POST /api/articles/:article_id/comments", () => {
...
+    return request(app)
+      .patch("/api/articles/104")
+      .send(updateVotes)
+      .expect(404)
+      .then((response) => {
+        expect(response.body.message).toBe("No such article_id exist yet");
+      });
+  });
+  it("status: 400, responds with an appropriate status and error when given invalid datatype for votes", () => {
```
Comment (nc-aimee-muter) [inline __tests__/api.test.js:246]: Where is your 400 test for an invalid article_id?

File: __tests__/api.test.js [hunk 5]
```diff
@@ -219,3 +219,39 @@ describe("/POST /api/articles/:article_id/comments", () => {
...
+    const falseVote = { inc_votes: "not-a-num" };
```
Comment (nc-aimee-muter) [inline __tests__/api.test.js:247]: What if inc_votes is not provided at all or if other properties are provided in addition to inc_votes? More tests needed in this describe block.

File: models/articles.model.js [hunk 6]
```diff
@@ -65,4 +69,33 @@ exports.insertCommentById = (article_id, username, body) => {
...
         });
     });
+}
+
+function incVotesById(article_id, inc_votes) {
+  return selectArticleById(article_id).then((data) => {
+    if (isNaN(inc_votes)) {
+      return Promise.reject({
+        status: 400,
+        message: "Cannot update. Invalid data.",
```
Comment (nc-aimee-muter) [inline models/articles.model.js:79]: You can be even more specific with this error message and really let the user know what is wrong.

File: __tests__/api.test.js (hunk 2 above)
Comment (zaw-tun) [reply __tests__/api.test.js:224]: Noted! I have implemented your feedback to be included my next pull request task 11. thank you!

File: __tests__/api.test.js (hunk 1 above)
Comment (zaw-tun) [reply __tests__/api.test.js:227]: Done! It will be included my next pull request task 11. thank you!

File: __tests__/api.test.js (hunk 4 above)
Comment (zaw-tun) [reply __tests__/api.test.js:246]: Sorry, I had the 404 test for invalid article_id. Has been using 404 for invalid article_ids in previous tests as well.

**Pull request merged**